src/System.Collections.Specialized/tests/HybridDictionary/GetItemObjTests.cs

```diff
@@ -14,40 +10,85 @@ public class GetItemObjTests
     {
         public const int MAX_LEN = 50;          // max length of random strings
 
+        private string[] TestValues(int length)
+        {
+            string[] values = new string[length];
+            for (int i = 0; i < length; i++)
+                values[i] = "Item" + i;
+            return values;
+        }
+
+        private string[] TestKeys(int length)
+        {
+            string[] keys = new string[length];
+            for (int i = 0; i < length; i++)
+                keys[i] = "keY" + i;
+            return keys;
+        }
+
+        private string[] Test_EdgeCases()
+        {
+            return new string[]
+            {
+                "",
+                " ",
+                "$%^#",
+                System.DateTime.Today.ToString(),
+                Int32.MaxValue.ToString()
+            };
+        }
+
+        [Theory]
+        [InlineData(50)]
+        [InlineData(100)]
+        public void Add(int size)
+        {
+            string[] values = TestValues(size);
+            string[] keys = TestKeys(size);
+
+            HybridDictionary hd = new HybridDictionary(true);
```

Comment:
I can't see any reason for ```hd.Add(keys[i].ToLower(), values[i].ToLower());``` to not instead be ```hd.Add(keys[i], values[i]);``` 

The Asserts should be ToLower and ToUpper for the keys, but I don't see why the values are also ToLowered. I guess to ensure that the map doesn't mess up the case of the values? That seems like an odd thing to test for, but I'm unfamiliar with the HybridDictionary.

I took this test from the test01 behemoth and modified it a bit to diagnose the intermittent failure issue (that ended up being something totally different). They could all use a refurbishing, but I'll leave that to https://github.com/dotnet/corefx/issues/754.